### ISSUE TITLE
video/tools/drawing.py  local variable 'norm' referenced before assignment

### ISSUE THREAD

**Issue by liikii** [Zulko/moviepy#1259]
# There seems to be a bug 

```
File "/home/debian2/anaconda3/lib/python3.7/site-packages/moviepy-2.0.0.dev1-py3.7.egg/moviepy/video/tools/drawing.py", line 156, in color_gradient
    n_vec = vector / norm ** 2  # norm 1/norm(vector)
UnboundLocalError: local variable 'norm' referenced before assignment
```

**Comment by tburrows13** [comment]
Thank you for the report. This is my fault: in #1101, the line `if vector is not None:` was incorrectly changed to `if vector:`, which caused #1205. Then I tried to fix it in #1212 but I apparently missed the `not`, so the logic is faulty. You can fix it in your installation by adding the `not` to that line: https://github.com/Zulko/moviepy/blob/master/moviepy/video/tools/drawing.py#L149

**Comment by tburrows13** [comment]
Fixed in https://github.com/Zulko/moviepy/commit/ff15fcfd2d211cbb2049f2eb2028e97739945b37
